client/src/visitors/pages/qrcode.js

```diff
@@ -108,54 +124,14 @@ export default class QRCode extends Component {
       });
   }
 
-  componentWillUpdate(nextProps, nextState) {
-    if (nextState.username === 'there is no registered user') {
```

Comment:
As we're no longer matching on this string, can this be removed from the route?

https://github.com/TwinePlatform/DataPower/blob/5dd8b49ab457c64245b50a7d5b54d6c70eb1ac88/server/routes/visitor_name.js#L18

Also in the above res it's sending the `hash: 0` property. Its also worth checking that the case this was previously solving, is also resolved in the rewrite.